src/iotconnect_telemetry.c

```diff
@@ -78,73 +78,48 @@ static cJSON *setup_telemetry_object(IotclMessageHandle message) {
-
-    // setup the actual telemetry object to be used in subsequent calls
-    message->current_telemetry_object = cJSON_CreateObject();
-    if (!message->current_telemetry_object) goto cleanup_da;
-
-    if (!cJSON_AddItemToArray(data_array, message->current_telemetry_object)) goto cleanup_cto;
-
-    return telemetry_object; // object inside the "d" array of the the root object
+    if (!cJSON_AddStringToObject(telemetry_object, "dt", time)) goto cleanup_to;
+    message->current_telemetry_object = cJSON_AddObjectToObject(telemetry_object, "d");
+    if (!message->current_telemetry_object) goto cleanup_cto;
+    if (!cJSON_AddItemToArray(message->telemetry_data_array, telemetry_object)) goto cleanup_cto;
+    char *sen = cJSON_Print(message->telemetry_data_array);
+    return telemetry_object; // object inside the "d" object of the the root object
 
     cleanup_cto:
     cJSON_free(message->current_telemetry_object);
-
-    cleanup_da:
-    cJSON_free(data_array);
-
     cleanup_to:
     cJSON_free(telemetry_object);
-
     return NULL;
 }
 
 IotclMessageHandle iotcl_telemetry_create(void) {
     cJSON *sdk_array = NULL;
     IotclConfig *config = iotcl_get_config();
     if (!config) return NULL;
-    if (!config->telemetry.dtg) return NULL;
     struct IotclMessageHandleTag *msg =
             (struct IotclMessageHandleTag *) calloc(sizeof(struct IotclMessageHandleTag), 1);
 
     if (!msg) return NULL;
-
     msg->root_value = cJSON_CreateObject();
-
     if (!msg->root_value) goto cleanup;
-
-    if (!cJSON_AddStringToObject(msg->root_value, "cpid", config->device.cpid)) goto cleanup_value;
-    if (!cJSON_AddStringToObject(msg->root_value, "dtg", config->telemetry.dtg)) goto cleanup_value;
-    if (!cJSON_AddNumberToObject(msg->root_value, "mt", 0)) goto cleanup_value; // telemetry message type (zero)
-    sdk_array = cJSON_AddObjectToObject(msg->root_value, "sdk");
-    if (!sdk_array) goto cleanup_value;
-    if (!cJSON_AddStringToObject(sdk_array, "l", CONFIG_IOTCONNECT_SDK_NAME)) goto cleanup_array;
-    if (!cJSON_AddStringToObject(sdk_array, "v", CONFIG_IOTCONNECT_SDK_VERSION)) goto cleanup_array;
-    if (!cJSON_AddStringToObject(sdk_array, "e", config->device.env)) goto cleanup_array;
+    if (!cJSON_AddStringToObject(msg->root_value, "dt", iotcl_iso_timestamp_now())) goto cleanup;   
 
     msg->telemetry_data_array = cJSON_AddArrayToObject(msg->root_value, "d");
-
+    
     if (!msg->telemetry_data_array) goto cleanup_array;
-
+    
     return msg;
 
     cleanup_array:
     cJSON_free(sdk_array);
-
-    cleanup_value:
-    cJSON_free(msg->root_value);
-
     cleanup:
     cJSON_free(msg);
     return NULL;
 }
 
 bool iotcl_telemetry_add_with_epoch_time(IotclMessageHandle message, time_t time) {
     if (!message) return false;
-    cJSON *telemetry_object = setup_telemetry_object(message);
+    cJSON *telemetry_object = setup_telemetry_object(message, time);
     if (!cJSON_AddNumberToObject(telemetry_object, "ts", time)) {
```

Comment:
It looks like setup_telemetry_object needs a (char*) for its second parameter -- may need to use ioctl_to_iso_timestamp(time) rather than using time directly?